app/main.py

```diff
@@ -1,3 +1,56 @@
+from __future__ import annotations
+from typing import Callable
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> None:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> None:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float | Callable) -> "Distance":
```

Comment:
```suggestion
    def __add__(self, other: int | float | Distance) -> "Distance":
```
imo, since other can be either Distance or number, it's better to use Distance annotation